pkgs/stdenv/generic/default.nix

```
@@ -152,9 +152,12 @@ let
         inherit lib config stdenv;
       }) mkDerivation;
 
-      # For convenience, bring in the library functions in lib/ so
-      # packages don't have to do that themselves.
-      inherit lib;
+      # Slated for deprecation in 21.11
+      lib = builtins.trace ( "Warning: `stdenv.lib` is deprecated "
+                           + "and will be removed in the next release. "
+                           + "Please use `pkgs.lib` instead. "
+                           + "For more information see https://github.com/NixOS/nixpkgs/issues/108938")
+                           lib;
```

```suggestion
      lib = builtins.trace
        ( "Warning: `stdenv.lib` is deprecated and will be removed in the next release."
         + " Please use `pkgs.lib` instead."
         + " For more information see https://github.com/NixOS/nixpkgs/issues/108938")
        lib;
```
I prefer that indentation never depend on the length of identifiers.